GenericJSON/Querying.swift

```diff
@@ -74,4 +74,33 @@ public extension JSON {
     subscript(dynamicMember member: String) -> JSON? {
         return self[member]
     }
+    
+    /// Return the JSON type at the keypath if this is an `.object`, otherwise `nil`
+    ///
+    /// This lets you write `json.valueForKeyPath("foo.bar.jar")`.
```

Comment:
```suggestion
    /// This lets you write `json[keyPath: "foo.bar.jar"]`.
```

Also, it would be good to give examples, but we can easily do that later after merging.

Reply:
Updated.